engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/NodeValidationExceptionHandler.scala

```diff
@@ -0,0 +1,31 @@
+package pl.touk.nussknacker.engine.compile
+
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{NonEmptyList, ValidatedNel}
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{CannotCreateObjectError, CustomNodeError, MissingParameters, NodeId}
+import pl.touk.nussknacker.engine.api.typed.{CustomNodeValidationException, MissingOutputVariableException}
+
+import scala.util.control.NonFatal
+
+object NodeValidationExceptionHandler {
+
+  def handleExceptions[T](f: => T)(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, T] = {
+    handleExceptionsInValidation(Valid(f))
+  }
+
+  def handleExceptionsInValidation[T](f: => ValidatedNel[ProcessCompilationError, T])(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, T] = {
+    try {
+      f
+    } catch {
+      case MissingOutputVariableException =>
+        Invalid(NonEmptyList.of(MissingParameters(Set("OutputVariable"), nodeId.id)))
+      case exc: CustomNodeValidationException =>
+        Invalid(NonEmptyList.of(CustomNodeError(exc.message, exc.paramName)))
+      case NonFatal(e) =>
```

Comment:
I know that, `FatalUnknownError` has no `nodeId` so those errors goes to "left corner" instead of "less-fatal" top of node ;). The message is `Could not create $nodeId: $message". It is IMO acceptable message also for services. WDYT?

Reply:
ok, it's fine for me